### PR TITLE
Add Admin Posting Details Page

### LINKED ISSUE #155: Posting details page for schedule creation
### Summary and Background
Admins can view more details for a particular posting when creating schedule.

### User Story
As an Admin, I want to view more details for a posting, in order to make a more informed decision on how to schedule volunteers .

### Solution (Core Elements)
From the volunteer scheduling page, the user can click the 'View posting' button to bring up more details on the specific posting.
The details include the following:

- role title / event name
- role description (purpose of the volunteer, outline of responsibilities, etc.)
- branch
- date/time
- relevant skills
- point of contact
- deadline to submit

If the admin wants to exit, they can select the 'Back to editing' button.

### Mockups

Other designs:
- https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Sistering-%2F-Admin-Experience?node-id=888%3A8374
- https://www.figma.com/file/KWaJRNJcOiW8ThZJ8Sl6On/Volunteer-Experience?node-id=157%3A3123.

Extremely similar to: https://app.zenhub.com/workspaces/sistering-614f8dad6afeb8001d428787/issues/uwblueprint/sistering/81

### Technical
* Refactor [VolunteerPostingDetails page](https://github.com/uwblueprint/sistering/blob/main/frontend/src/components/pages/volunteer/posting/VolunteerPostingDetails.tsx):
    * Core details card should be moved into a separate component in `frontend/src/components/common/PostingDetails.tsx`
    * The updated VolunteerPostingDetails page should render the "Back to volunteer postings" and "Submit availabilities" header links, then the above component, then the deadline and "Submit availabilities" footer
* Create a new page called AdminPostingDetails under `frontend/src/components/pages/admin/posting/AdminPostingDetails.tsx ` with route `/admin/posting/:id`. This page renders a "Back to editing" header link, and then the PostingDetails component

### PR DESCRIPTION
**Pull request by albertlai431**
## Ticket link

Closes #155 

## Implementation description
* Refactored VolunteerPostingDetails page
* Created routing for AdminPostingDetails page
* Created query to fetch Posting data and the associated Admin Posting Details Page UI

## Steps to test
1. `docker-compose up --build`
2. Navigate to http://localhost:3000/admin/posting/1 and verify that the UI matches that specified in #155 

## What should reviewers focus on?
* Verify that the query works and the UI matches that specified in #155

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/volunteer/posting/VolunteerPostingDetails.tsx [hunk 1]
```diff
@@ -74,58 +73,7 @@ const VolunteerPostingDetails = (): React.ReactElement => {
...
-                    <PocCard
-                      name="John Doe"
-                      title="hard code"
-                      email="hard code"
-                      phoneNumber="hard code"
-                      key={1}
-                    />
-                  </HStack>
+                  <PostingDetails postingDetails={postingDetails} />
                   <Divider />
```
Comment (LenaNguyen) [inline frontend/src/components/pages/volunteer/posting/VolunteerPostingDetails.tsx:77]: Can you check with design to see if the bottom part of the posting details should be included for the **admin** view? We might need to include the deadline at the bottom but without the button.

If so, we should include this bottom part in the `PostingDetails` component and add a string prop like `footerButtonText`. If the prop is null, then don't render the button. 

image

**Review by LenaNguyen — CHANGES_REQUESTED** [review]
Just one small thing!

File: frontend/src/components/pages/volunteer/posting/VolunteerPostingDetails.tsx (hunk 1 above)
Comment (albertlai431) [reply frontend/src/components/pages/volunteer/posting/VolunteerPostingDetails.tsx:77]: Yep, just confirmed with @krystaltruong that the footer for the admin view should contain the deadline but not the button! I'll make the requested changes.

**Review by LenaNguyen — APPROVED** [review]

File: frontend/src/components/pages/admin/posting/AdminPostingDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import { VStack, Box, Container, Button, Flex } from "@chakra-ui/react";
+
+import { gql, useQuery } from "@apollo/client";
+import { useParams } from "react-router-dom";
+import { ChevronLeftIcon } from "@chakra-ui/icons";
+import { PostingResponseDTO } from "../../../../types/api/PostingTypes";
+import PostingDetails from "../../../common/PostingDetails";
+
+const POSTING = gql`
+  query Posting($id: ID!) {
```
Comment (sherryhli) [inline frontend/src/components/pages/admin/posting/AdminPostingDetails.tsx:11]: Small suggestion: let's prefix all operation names with the component they're located in, this is a common practice with GraphQL as it helps with debugging

```suggestion
  query AdminPostingDetails_Posting($id: ID!) {
```

Comment (albertlai431) [reply frontend/src/components/pages/admin/posting/AdminPostingDetails.tsx]: Done!

**Review by sherryhli — APPROVED** [review]
Awesome, LGTM!

**Pull request merged**